src/frontends/pytorch/src/op/index_put_.cpp

```diff
@@ -24,101 +12,13 @@ namespace op {
-        }
-        broadcast_index_shape = context.mark_node(std::make_shared<v3::ShapeOf>(index, element::i32));
-        OutputVector indices_list;
-        for (int i = 0; i < indices_list_len; i++) {
-            auto broadcast =
-                context.mark_node(std::make_shared<v3::Broadcast>(split_indices->output(i), broadcast_index_shape));
-            auto unsqueeze = context.mark_node(std::make_shared<v0::Unsqueeze>(broadcast, const_neg_1));
-
-            // change negative indices to positive indices
-            auto const_i = context.mark_node(v0::Constant::create(element::i32, Shape{}, {i}));
-            auto dim_i = context.mark_node(std::make_shared<v8::Gather>(input_shape, const_i, const_0));
-            auto dim_i_correct_type = context.mark_node(std::make_shared<v1::ConvertLike>(dim_i, index));
-            unsqueeze = context.mark_node(std::make_shared<v1::Add>(unsqueeze, dim_i_correct_type));
-            unsqueeze = context.mark_node(std::make_shared<v1::Mod>(unsqueeze, dim_i_correct_type));
-
-            indices_list.push_back(unsqueeze);
-        }
-        index = context.mark_node(std::make_shared<v0::Concat>(indices_list, -1));
-    } else {
-        index = split_indices->output(0);
-
-        // change negative indices to positive indices
-        auto dim_0 = context.mark_node(std::make_shared<v8::Gather>(input_shape, const_0, const_0));
-        auto dim_0_correct_type = context.mark_node(std::make_shared<v1::ConvertLike>(dim_0, index));
-        index = context.mark_node(std::make_shared<v1::Add>(index, dim_0_correct_type));
-        index = context.mark_node(std::make_shared<v1::Mod>(index, dim_0_correct_type));
-
-        broadcast_index_shape = context.mark_node(std::make_shared<v3::ShapeOf>(index, element::i32));
-        index = context.mark_node(std::make_shared<v0::Unsqueeze>(index, const_neg_1));
-    }
-
-    auto sub_data_shape =
-        context.mark_node(std::make_shared<v8::Slice>(input_shape, const_indices_list_len, const_max_int, const_1));
-    auto values_shape =
-        context.mark_node(std::make_shared<v0::Concat>(OutputVector{broadcast_index_shape, sub_data_shape}, 0));
-    values = context.mark_node(std::make_shared<v3::Broadcast>(values, values_shape));
-    values = context.mark_node(std::make_shared<v1::ConvertLike>(values, input));
-
-    Output<Node> result;
-    if (accumulate) {
-        auto zeros = generate_zeros_with_convertlike(context, input_shape, input);
-        result = context.mark_node(std::make_shared<v3::ScatterNDUpdate>(zeros, index, values));
-        result = context.mark_node(std::make_shared<v1::Add>(input, result));
-    } else {
-        result = context.mark_node(std::make_shared<v3::ScatterNDUpdate>(input, index, values));
-    }
-
-    return {result};
+    // Pass as PtFrameworkNode to register as `inplace_op`. Conversion to OV operators is done as transformation.
```

Comment:
In fact this is not required, regular fw node creation should detect the fact that the operation is modifying input if it has "!" in schema. But it is okay to leave this translation here. Also I would prefer that custom translator is not marked as `inplace_op`, it seems much less obvious what is happen during translation.

Reply:
Removing it from op_table was my first idea, however, it seem to fail. Creating simple fw_node was the simplest way I though for fixing it, didn't known that there is more generic approach, I'll check it and look what went wrong.

Reply:
The generic approach may fail if schema doesn't contain "!". It makes sense to have translator in that case.

Reply:
I've checked that schema contains "!" and code works correctly, issue with number of outputs throw me out, however, now I see that it may be intended [since it add one additional output](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/pytorch/src/utils.cpp#L172). Should then transformation replace both outputs with the same node?

Reply:
Ah yeah, that is the downside of generic approach, in your case it would be better to have separate translator to not handle one extra output

Reply:
Yeah, since there is also out-place version of aten::index_put_, I guess that adding dummy translation and using inplace_op might be more generic since then transformation should be able to handle both uses